src/utils/dates.js

```diff
@@ -10,3 +10,15 @@ export const ONE_DAY_IN_MILLISECONDS = 86400000;
 export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
+
+// This function will enable us to convert JS dates into Firebase-friendly timestamps, calculate the difference, and then return the difference between them (as a number).
+
+export function getDaysBetweenDates(firstDate, secondDate) {
+	// Takes two JS Date objects, converts to MS, calculates MS difference, then converts to days as a number (rounded up or down).
+	const firstTime = firstDate.getTime();
+	const secondTime = secondDate.getTime();
+	const secondsBetween = secondTime - firstTime;
+	const daysBetween = secondsBetween / ONE_DAY_IN_MILLISECONDS;
+	// Round number here
+	return Math.round(daysBetween);
```

Comment:
I want to point out that `Math. round` here, it will not be consistent in terms of rounding.
If the time difference is less than 12 hours, it will round to zero, but if it's more than 12 hours, it will round up to the next hour.

In my opinion, it should be used as `Math.floor`, but it will depend on the requirement of the calculation.

![CleanShot 2024-03-09 at 21 10 28](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/22878284/ead9b001-2719-41ec-bb09-eb8072bcdc4d)



![CleanShot 2024-03-09 at 21 09 04](https://github.com/the-collab-lab/tcl-68-smart-shopping-list/assets/22878284/bf797097-628c-406f-b4bc-1eb015236b60)